### PR TITLE
RR-544 - Add goals from pre-induction overview page

### PR DESCRIPTION
**Pull request by nathanrussell-moj-digital**
This PR implements RR-544 to add functionality to the pre-induction Overview screen to allow adding Goals before the CIAG Induction is created (and subsequently displaying those goals on the pre-induction Overview screen)

No changes have been made to the post-induction Overview screen of process flow.

https://github.com/ministryofjustice/hmpps-education-and-work-plan-ui/assets/94835226/e413d31b-51ee-4c23-adf5-86a4e88ab526

### CODE REVIEW

File: helm_deploy/values-dev.yaml [hunk 1]
```diff
@@ -24,6 +24,7 @@ generic-service:
     DPS_URL: "https://digital-dev.prison.service.justice.gov.uk"
     FRONTEND_COMPONENT_API_URL: "https://frontend-components-dev.hmpps.service.justice.gov.uk"
     ENVIRONMENT_NAME: "DEV"
+    CREATE_GOALS_WITHOUT_INDUCTION_ENABLED: "true"
```
Comment (nathanrussell-moj-digital) [inline helm_deploy/values-dev.yaml:27]: Feature toggled; only enabled in dev at the moment

File: server/testsupport/actionPlanTestDataBuilder.ts [hunk 2]
```diff
@@ -1,6 +1,34 @@
...
+ * ```
+ * // create an action plan with 3 goals
+ * const actionPlan = aValidActionPlanWithOneGoal(
+ *  {
+ *    goals: [aValidGoal(), aValidGoal(), aValidGoal()],
+ *  }
+ * )
+ * ```
+ */
 const aValidActionPlanWithOneGoal = (
```
Comment (nathanrussell-moj-digital) [inline server/testsupport/actionPlanTestDataBuilder.ts:32]: In this PR I decided to deprecate this method with a suitable comment rather than rename it, otherwise this PR would have a load of unrelated change where I'd renamed this method.
We can remove this deprecated method and change all the existing uses of it to `aValidActionPlan()` in a separate PR

File: server/views/pages/overview/partials/overviewTab/overviewTabPostInduction.njk [hunk 3]
```diff
@@ -3,103 +3,8 @@
 <section data-qa="post-induction-overview">
   {% if not actionPlan.problemRetrievingData %}
 
-    <div class="moj-page-header-actions">
```
Comment (nathanrussell-moj-digital) [inline server/views/pages/overview/partials/overviewTab/overviewTabPostInduction.njk:6]: I've pulled out the action button header and goals list into includes (above) because they will also now be used in the pre-induction overview page (below)

File: server/views/pages/overview/partials/overviewTab/overviewTabPreInduction.njk [hunk 4]
```diff
@@ -1,16 +1,38 @@
...
+    {{ govukNotificationBanner({
+      html: createInductionBanner
+    }) }}
+
+  {% else %}
+    {% if not actionPlan.problemRetrievingData %}
+
+      {% include './_actionsHeader.njk' %}
+
+      {{ govukNotificationBanner({
```
Comment (nathanrussell-moj-digital) [inline server/views/pages/overview/partials/overviewTab/overviewTabPreInduction.njk:24]: Put the banner between the action buttons header and the goals list, as per the design

**Review by chrimesdev — APPROVED** [review]
LGTM

File: server/testsupport/actionPlanTestDataBuilder.ts (hunk 2 above)
Comment (mattwills-moj-digital) [reply server/testsupport/actionPlanTestDataBuilder.ts:32]: I may have missed something, but `aValidActionPlanWithOneGoal()` seems ok to me?! Perhaps the goals array argument could be changed to a single `goal`, so it can be changed by a test if necessary, but otherwise I think it's ok! :)

**Review by mattwills-moj-digital — DISMISSED** [review]

File: integration_tests/e2e/overview/preInductionOverview.cy.ts [hunk 5]
```diff
@@ -13,14 +14,18 @@ context('Prisoner Overview page - Pre Induction', () => {
...
     cy.task('stubGetCiagProfile404Error')
-    cy.task('getActionPlan')
-    cy.task('stubLearnerProfile')
-    cy.task('stubLearnerEducation')
+    cy.task('getActionPlan', 'G6115VJ')
+    cy.task('getActionPlan', 'A00001A')
+    cy.task('stubLearnerProfile', 'G6115VJ')
+    cy.task('stubLearnerProfile', 'A00001A')
+    cy.task('stubLearnerEducation', 'G6115VJ')
+    cy.task('stubLearnerEducation', 'A00001A')
```
[the remark this answers is not in the document]
Comment (nathanrussell-moj-digital) [reply integration_tests/e2e/overview/preInductionOverview.cy.ts:25]: Yeah, agreed. I think many of our integration tests could do with looking at in this respect because they all setup stuff like this. I'm sure there's a way of reducing the noise and duplication across all of our tests 👍 
Like you say, one for another day in another PR 👍

File: server/testsupport/actionPlanTestDataBuilder.ts (hunk 2 above)
Comment (nathanrussell-moj-digital) [reply server/testsupport/actionPlanTestDataBuilder.ts:32]: I think that was the problem though - the method name says its an action plan with one goal, but the method arg allows you to specify any number of goals. And I was after setting up an action plan with no goals, so I really didnt want to use this method.

But you are right that instead of deprecating the method we could change the method arg to only take a single goal. Leave it with me  ..... 👍
